src/app/CMakeLists.txt

```
@@ -10,13 +10,33 @@ set(QML_IMPORT_PATH "${CMAKE_SOURCE_DIR}/src/qml/imports")
 file(GLOB_RECURSE QML_SOURCES "${CMAKE_SOURCE_DIR}/src/qml/*.qml")
 add_custom_target(qfield_qml SOURCES ${QML_SOURCES})
 
+# lrelease .ts files
+find_package(Qt5 COMPONENTS LinguistTools)
+
+file(GLOB TS_FILES ${CMAKE_SOURCE_DIR}/i18n/*.ts)
+set_source_files_properties(${TS_FILES} PROPERTIES OUTPUT_LOCATION "${CMAKE_SOURCE_DIR}/i18n")
```

```suggestion
set_source_files_properties(${TS_FILES} PROPERTIES OUTPUT_LOCATION "${CMAKE_BINARY_DIR}/i18n")
```
if you want to have them in build directory

---

@3nids , I've tried that, without success so far.

---

from the docs 

> By default, the qm files will be placed in the root level of the build directory. To change this, you can set OUTPUT_LOCATION as a property of the source .ts file.

if you only write `set_source_files_properties(${TS_FILES} PROPERTIES OUTPUT_LOCATION "i18n")`, what happens?

---

folder gets created but no qm inside


ninja: error: 'i18n/qfield_sv.qm', needed by 'src/app/qfield_autogen/P4RBTXXXTD/qrc_translations.cpp', missing and no known rule to make it
20:16:08: The process "/usr/bin/cmake" exited with code 1.
Error while building/deploying project QField (kit: Desktop)

---

and nothing anywhere?

---

you can maybe copy the Qt macro https://github.com/qt/qttools/blob/5.15.2/src/linguist/Qt5LinguistToolsMacros.cmake#L112
and output some debug messages to see what's going on